src/Language/Spectacle/Syntax/Modal.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE TupleSections #-}
+
+module Language.Spectacle.Syntax.Modal
+  ( Modal (Modal),
+    Effect (Always, UpUntil),
+    always,
+    eventually,
+    upUntil,
+    testModalResult1,
+  )
+where
+
+import Control.Monad
```

Comment:
Explicitly enumerating all of the import bindings or qualifying the modules will improve maintainability. This comment goes for all of the modules that use implicit import scope.